dashboard-app/src/dashboard/ByActivity/__tests__/datePickerConstraints.test.ts

```diff
@@ -0,0 +1,63 @@
+import moment from 'moment';
+import DatePickerConstraints from '../datePickerConstraints';
+
+
+describe('Activity view date-picker constraints', () => {
+  describe('from', () => {
+    test('No dates before Jan 2017', () => {
+      const date = DatePickerConstraints.from.min(new Date, new Date); // inputs unimportant here
+      expect(moment('2017-01-01').isSame(date)).toBe(true);
+    });
+    test('No future dates', () => {
+      const date = DatePickerConstraints.from.max(new Date, new Date); // inputs unimportant here
+      expect(moment().startOf('day').isSameOrBefore(date)).toBe(true);
+    });
+    test('from < to is OK: transformed to start of day', () => {
+      const from = moment().subtract(1, 'day');
+      const to = moment();
+      const date = DatePickerConstraints.from.validate(from.toDate(), to.toDate());
+      expect(from.startOf('day').isSame(date)).toBe(true);
+    });
+    test('from > to is OK: transformed to start of day', () => {
```

Comment:
Will add a comment. The reason the test is named that way is because the validation is on the `from` date _only_. With our current rules, this date is always valid.